sdk/kiosk/src/tx/kiosk.ts

```diff
@@ -200,20 +203,16 @@ export function purchase(
   tx: TransactionBlock,
   itemType: string,
   kiosk: ObjectArgument,
-  itemId: SuiAddress,
+  item: ObjectArgument,
```

Comment:
Here we only need itemId, ObjectArgument doesn't really make sense since you can't access the item object directly before it has been purchased / taken.